eth/v1/beacon_state.proto

```diff
@@ -0,0 +1,147 @@
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/finality_checkpoints" };
+  }
+
+  rpc Validators(StateRequest) returns (StateValidatorsResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/validators" };
+  }
+
+  rpc Validator(StateValidatorRequest) returns (StateValidatorResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/validators/{validator_id}" };
+  }
+
+  rpc Committees(StateCommitteesRequest) returns (StateCommitteesResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/committees/{epoch}" };
+  }
+}
+
+message GenesisResponse {
+  string genesis_time = 1;
+  string genesis_validators_root = 2;
+  string genesis_fork_version = 3;
+}
+
+message StateRequest {
+  // The state id which can be any of: "head" (canonical head in node's view),
+  // "genesis", "finalized", "justified", <slot>, <hex encoded stateRoot with 0x prefix>.
+  // Uses the provided state_id for the request.
+  string state_id = 1;
+}
+
+message StateRootResponse {
+  string state_root = 1;
+}
+
+message StateForkResponse {
+  Fork fork = 1;
+}
+
+message Fork {
+  string previous_version = 1;
+  string current_version = 2;
+  string epoch = 3;
+}
+
+message StateFinalityCheckpointResponse {
+  Checkpoint previous_justified = 1;
+  Checkpoint current_justified = 2;
+  Checkpoint finalized = 3;
+}
+
+message Checkpoint {
+  string epoch = 1;
+  string root = 2;
+}
+
+message StateValidatorsResponse {
+  repeated ValidatorContainer validators = 1;
+}
+
+message StateValidatorRequest {
+  // The state id which can be any of: "head" (canonical head in node's view),
+  // "genesis", "finalized", "justified", <slot>, <hex encoded stateRoot with 0x prefix>.
+  // Uses the provided state_id for the request.
+  string state_id = 1;
+  // The public key or index for the validator to retrieve information for.
+  string validator_id = 2;
+}
+
+message StateValidatorResponse {
+  ValidatorContainer validator = 1;
+}
+
+message ValidatorContainer {
+  string index = 1;
+  string balance = 2;
+  string status = 3;
+  Validator validator = 4;
+}
+
+message Validator {
+  string pubkey = 1;
+  string withdrawal_credentials = 2;
+  string effective_balance = 3;
+  bool slashed = 4;
+  string activation_eligibility_epoch = 5;
+  string activation_epoch = 6;
+  string exit_epoch = 7;
+  string withdrawable_epoch = 8;
+}
+
+message StateCommitteesRequest {
+  // The state id which can be any of: "head" (canonical head in node's view),
+  // "genesis", "finalized", "justified", <slot>, <hex encoded stateRoot with 0x prefix>.
+  // Uses the provided state_id for the request.
+  string state_id = 1;
+  // The epoch to retrieve the committees of.
+  string epoch = 2;
+}
+
+message StateCommitteesResponse {
+  repeated Committee committee = 1;
+}
+
+message Committee {
+  string index = 1;
+  string slot = 2;
+  repeated string validators = 3;
```

Comment:
are these validator indices? maybe rename to validator_indices and make it a repeated uint64 field

Reply:
We might not be able to rename though :(. If that's the expected response name then we have no choice